route53.tf

```diff
@@ -1,15 +1,28 @@
+locals {
+  alias = [for a in var.alias : {
+    domain       = a.domain
+    hostname     = a.hostname
+    private_zone = lookup(a, "private_zone", false)
+    vpc_id       = lookup(a, "private_zone", false) ? data.aws_vpc.selected.id : null
+    ttl          = lookup(a, "ttl", 60)
+    }
+  ]
+}
+
 data "aws_route53_zone" "selected" {
   count = length(var.alias)
 
-  name = lookup(var.alias[count.index], "domain", "")
+  name         = local.alias[count.index].domain
```

Comment:
The purpose of the added local variable was to facilitate the addition of the VPC id for private zones in a way that is reasonably readable. Perhaps there are other ways to accomplish that – do you have ideas?

While creating a bunch of intermediate variables sometimes annoys me, it's something I had to do heavily in the `enterprise-subnets` module to be generic and to use sophisticated data structures without making the code a complete nightmare of nested maps and lists that no one including me will be able to read three months from now. :-)

I don't think sheer efficiency [by avoiding temporary variables] is worth fretting about in Terraform code, and I tend to err on the side of readability vs. efficiency.

Reply:
@ddriddle, honestly this is why I like reviewing each other's code. I added the logic I did because at first glance, the addition of private zone support looked more complicated than it actually was. And when I finished, I was tired enough that I forgot to take a step back and see that the extra data structure wasn't needed.

I've refactored the `local.alias` out of the module.